src/asl.ts

```diff
@@ -4058,7 +4058,7 @@ export namespace ASLGraph {
       .filter(ASLGraph.isStateOrSubState);
     return realStates.length === 0
       ? undefined
-      : realStates.length === 1
+      : realStates.length === 1 && realStates[0]
       ? { node, ...realStates[0] }
```

Comment:
I think it might be better in this case to do `...readStates[0]!` since our contract is such that the items in the array are never `undefiend`?